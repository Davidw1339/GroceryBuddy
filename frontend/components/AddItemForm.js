import React from 'react';
import { View, StyleSheet, ScrollView, KeyboardAvoidingView, Image } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { Input, Text, Button } from 'react-native-elements';
import { addGroceryItem } from '../utils/api';
import { ImageManipulator } from 'expo';

const INPUT_FIELDS = [
    {
        name: "name",
        label: "Name",
        placeholder: "Apples"
    },
    {
        name: "upc",
        label: "UPC",
        keyboardType: 'numeric',
        placeholder: "00000000"
    },
    {
        name: "price",
        label: "Price",
        keyboardType: 'numeric',
        placeholder: "3.42"
    },
    {
        name: "store",
        label: "Store",
        placeholder: "County Market",
    },
    {
        name: "lat",
        label: "Latitude",
        keyboardType: 'numeric',
        placeholder: "40.113264",
    },
    {
        name: "long",
        label: "Longitude",
        keyboardType: 'numeric',
        placeholder: "-88.233905",
    }
]

export default class AddItemForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = { 
      name: '',
      upc: '',
      price: '',
      store: '',
      user: 'admin',
      image: '',
      image_uri: '',
      lat: '',
      long: ''
    }
  }

  retrievePicture = (image) => {
    // Manipulate the image by resizing it to a smaller size
    ImageManipulator.manipulateAsync(image.uri, [{resize: {width: 300}}], {base64: true})
    .then((image) => {
      const base64_datauri = 'data:image/jpg;base64,' + image.base64;
      // set the state correctly so that the data can be sent to the backend
      this.setState({image_uri: base64_datauri, image: image.base64});
    })
  }

  openCamera = () => {
    this.props.navigation.navigate('Camera', {
      handlePictureTaken: this.retrievePicture
    })
  }

  retrieveUPC = (type, data) => {
    this.setState({
      upc: data 
    });
  }

  openScanner = () => {
    this.props.navigation.navigate('Scanner', {
      handleBarCodeScanned: this.retrieveUPC
    });
  }

  submitItemForm = () => {
    console.log(this.state);
    addGroceryItem(this.state);
  }

  render() {
    return (
      <ScrollView>
        <KeyboardAvoidingView behavior="position" enabled>
          <View style={styles.container}>
            <Text h4 style={styles.input}>Add A Grocery Item</Text>
            {this.state.image_uri !== "" && (<Image style={[styles.image, styles.input]} source={{uri: this.state.image_uri}}/>)}
            <Button
              title="Take A Picture"
              raised
              onPress={this.openCamera}
              />
            {INPUT_FIELDS.map((fieldProps) => {
              return <Input
                  key={fieldProps.name}
                  onChangeText={(textInput) => this.setState({ [fieldProps.name]: textInput })}
                  value={this.state[fieldProps.name]}
                  containerStyle={styles.input}
                  {...fieldProps}
                  />
            })}
            <Button
              title="  Scan UPC"
              raised
              icon={<Ionicons name="ios-camera" size={24} color={"white"}/>}
              onPress={this.openScanner}
              containerStyle={styles.input}
              />
            <Button
              title="Submit Item"
              raised
              onPress={this.submitItemForm}
              />
          </View>
        </KeyboardAvoidingView>
      </ScrollView>
    );
  }
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    paddingTop: 20
  },
  input: {
    marginBottom: 20
  },
  image: {
    width: 200,
    height: 200
  }
});