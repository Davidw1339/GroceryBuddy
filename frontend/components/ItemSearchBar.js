import React from 'react';
import { StyleSheet, View } from 'react-native';
import { SearchBar, Button } from 'react-native-elements';
import { Ionicons } from '@expo/vector-icons';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    borderBottomColor: 'lightgray',
    borderBottomWidth: 1,
  },
  input: {
    flex: 1,
  },
  scanbutton: {
    height: 35,
  },
});

export default class ItemSearchBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      search: '',
    };
  }

  /**
   * renders search bar
   *
   * @return rendered search bar
   */
  render() {
    return (
      <View style={styles.container}>
        <SearchBar
          platform="android"
          placeholder="Grocery Item Name Here"
          onChangeText={search => this.setState({ search })}
          value={this.state.search}
          onSubmitEditing={() => this.props.onSearch(this.state.search)}
          containerStyle={styles.input}
        />
        <Button
          raised
          icon={<Ionicons name="ios-camera" size={24} color="white" />}
          onPress={this.props.onPressCamera}
          containerStyle={styles.scanbutton}
        />
      </View>
    );
  }
}
