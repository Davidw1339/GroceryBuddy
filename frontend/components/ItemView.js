import React from 'react';
import {
  StyleSheet,
  View,
  Text,
} from 'react-native';

const styles = StyleSheet.create({
  container: {
    marginTop: 65,
  },
  heading: {
    backgroundColor: '#F8F8F8',
  },
  name: {
    fontSize: 25,
    fontWeight: 'bold',
    margin: 5,
    color: '#656565',
  },
  price: {
    fontSize: 20,
    fontWeight: 'bold',
    margin: 5,
    color: '#48BBEC',
  },
  title: {
    fontSize: 20,
    margin: 5,
    color: '#656565',
  },
});

export default class ItemView extends React.Component {
  static navigationOptions = {
    title: 'Item Details',
  }

  /**
   * renders item with price
   *
   * @return rendered item
   */
  render() {
    const { params } = this.props.navigation.state;
    const { item } = params;
    const { name } = item;
    const price = `$${item.price}`;

    return (
      <View style={styles.heading}>
        <Text style={styles.name}>{name}</Text>
        <Text style={styles.price}>{price}</Text>
      </View>
    );
  }
}
