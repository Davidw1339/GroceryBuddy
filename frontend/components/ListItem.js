import React from 'react';
import {
  StyleSheet, View, TouchableOpacity, TouchableNativeFeedback, TouchableHighlight, Platform,
} from 'react-native';
import { Card, Text } from 'react-native-elements';
import Icons from '@expo/vector-icons/FontAwesome';

const styles = StyleSheet.create({
  container: {
    marginTop: 65,
  },
  card: {
    padding: 10,
    borderRadius: 10,
  },
  cardControlContainer: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  cardControl: {
    margin: 10,
  },
  cardText: {
    flex: 1,
    marginLeft: 20,
    fontSize: 18,
    fontWeight: 'bold',
  },
  cardContentContainer: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },
});

/**
 * ListItem describes a clickable list element that is displayed on the homepage
 */
const ListItem = (props) => {
  const TouchablePlatformSpecific = Platform.OS === 'ios'
    ? TouchableHighlight
    : TouchableNativeFeedback;
  return (
    <TouchablePlatformSpecific onPress={props.handleLaunchList}>
      <Card containerStyle={styles.card}>
        <View style={styles.cardContentContainer}>
          <Text style={styles.cardText}>{props.name}</Text>
          <View style={styles.cardControlContainer}>
            <TouchableOpacity style={styles.cardControl}>
              <Icons name="pencil" size={24} color="gray" />
            </TouchableOpacity>
            <TouchableOpacity style={styles.cardControl} onPress={props.handleDelete}>
              <Icons name="trash-o" size={24} color="gray" />
            </TouchableOpacity>
          </View>
        </View>
      </Card>
    </TouchablePlatformSpecific>
  );
};

export default ListItem;
