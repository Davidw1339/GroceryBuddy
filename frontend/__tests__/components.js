


// __tests__/Intro-test.js
import React from 'react';

import renderer from 'react-test-renderer';
import AddItemForm from "../components/AddItemForm";
import ItemSearchBar from "../components/ItemSearchBar";
import ItemView from "../components/ItemView";
import ListItem from "../components/ListItem";
import ItemAdditionForm from "../components/UPCScanner";

test('AddItemForm', () => {
    jest.useFakeTimers();
    const tree = renderer.create(<AddItemForm />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('ItemSearchBar', () => {
    jest.useFakeTimers();
    const tree = renderer.create(<ItemSearchBar />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('ListItem', () => {
    jest.useFakeTimers();
    const tree = renderer.create(<ListItem />).toJSON();
    expect(tree).toMatchSnapshot();
});


test('UPCScanner', () => {
    jest.useFakeTimers();
    const tree = renderer.create(<ItemAdditionForm />).toJSON();
    expect(tree).toMatchSnapshot();
});

