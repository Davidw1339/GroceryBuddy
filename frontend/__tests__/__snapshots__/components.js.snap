// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItemForm: AddItemForm 1`] = `null`;

exports[`ItemSearchBar: ItemSearchBar 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "lightgray",
      "borderBottomWidth": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "paddingBottom": 8,
        "paddingTop": 8,
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 0,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 0,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginLeft": 8,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
              underlayColor="white"
            >
              <Text />
            </View>
          </View>
        </View>
        <TextInput
          allowFontScaling={true}
          onBlur={[Function]}
          onCancel={[Function]}
          onChangeText={[Function]}
          onClear={[Function]}
          onFocus={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Grocery Item Name Here"
          platform="android"
          style={
            Object {
              "alignSelf": "center",
              "color": "black",
              "flex": 1,
              "fontSize": 18,
              "marginLeft": 24,
              "marginRight": 8,
              "minHeight": 40,
            }
          }
          testID="searchInput"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginLeft": 15,
              "marginRight": 8,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 3,
        "height": 35,
        "shadowColor": "rgba(0,0,0, .4)",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#2089dc",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
</View>
`;

exports[`ListItem: ListItem 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "#e1e8ee",
        "borderRadius": 10,
        "borderWidth": 1,
        "margin": 15,
        "marginBottom": 0,
        "padding": 10,
        "shadowColor": "rgba(0,0,0, .2)",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "flex": 1,
              "fontSize": 18,
              "fontWeight": "bold",
              "marginLeft": 20,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "margin": 10,
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "margin": 10,
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`UPCScanner: UPCScanner 1`] = `
<Text>
  Requesting for camera permission
</Text>
`;
